src/status_im/ui/screens/offline_messaging_settings/events.cljs

```diff
@@ -16,6 +18,45 @@
                               {:dispatch [:logout]}
                               (accounts-events/update-settings (assoc-in settings [:wnode network] wnode))))))
 
+(handlers/register-handler-fx
+ :save-wnode-transaction
+ (fn [{:keys [db] :as cofx} [_ wnode hash]]
+   (let [network     (ethereum/network->chain-keyword (:network db))
+         settings    (get-in db [:account/account :settings])
+         transaction (get-in db [:wallet :transactions hash])]
+     (when (not (nil? transaction))
+       (handlers-macro/merge-fx cofx
+                                {:dispatch [::save-wnode wnode hash]}
+                                (accounts-events/update-settings (assoc-in settings [:wnode-payment network wnode] hash)))))))
+
+(handlers/register-handler-fx
+ ::pay-wnode
+ (fn [{{:keys [web3] :as db} :db} [_ wnode]]
+   (let [network                         (ethereum/network->chain-keyword (:network db))
+         {:keys [address amount symbol]} (get-in db [:inbox/wnodes network wnode :payment])
+         contract                        (:address (tokens/symbol->token network symbol))
+         from                            (get-in db [:account/account :address])
+         gas                             (ethereum/estimate-gas symbol)
+         gas-price                       (money/->wei :gwei 5)]
+     {:db (update-in db [:wallet :send-transaction] assoc :waiting-signal? false)}
```

Comment:
This looks a bit odd to me, unless I'm reading it wrong it looks like two statements:

`{:db ...}`
`(erc20/transfer ...)`

which would imply first is imperative. Is this supposed to be wrapped in an if-statement, or are they both supposed to be in a single map perhaps?

Reply:
I am curious what the idiomatic way is for updating the db then calling the `transfer` function. I have removed the `{:db }` and now call update directly (not entirely sure if this fixed it?)